contracts/new_StakeVault.sol

```diff
@@ -57,16 +66,23 @@ contract StakeVault is Ownable {
     ) public {}
 
     function unStake(uint256 stakeId) public {}
-
-    function commitStake(uint256 stakeId) public {}
+    
+    function commitStake (uint256 stakeId) public 
+        senderOwnsStake(stakeId)
+    {
+        require(!_stakes[stakeId].isCommitted, "Stake is already committed");
+        require(_stakes[stakeId].staker != address(0), "Stake doesn't exist");
```

Comment:
Why are we checking if the investor exists?

The message implies that it is validation the stake but it is acually validating the investor account